nf-core/scrnaseq
# update readme

- Bump to 2.1.1dev
- Removed function get_samplesheet_paths which is replicated in subworkflows/local/input_check.nf
- Samplesheet documentation includes expected cells and sequencing center
- Validation of optional samplesheet columns expected_cells and seq_center
- Prettier & PythonBlack checks
- PythonBlack fix
- Clarifying that params.seq_center can be overridden by setting seq_center in samplesheet
- Allows for sequencing center to be set in samplesheet for STARsolo BAM file read group info
- Prettier fix
- Added expected cells to STARsolo via --soloCellFilter
- Linting fix
- Addition of --expect-cells flag using samplesheet input
- Added optional expected cells parameter to cellranger counts process
- Linting fixes for spaces
- autocanceling previous tests when pushing changes
- triggering change to test autocancel
- updated changelog
- Added required/optional to samplesheet columns in usage
- Fix CI
- Update ci.yml
- [automated] Fix linting with Prettier
- updated concurrency groups to match nf-core tools
- Updated simpleaf to version 0.5.3 to allow for expect-cells parameter
- Add stub for versions.yml in mtx_to_seurat configuration
- Add single-nucleus, RNA velocity option on star.
- Adding svg for pipeline picture
- Adding current version of scrnaseq pipeline picture
- renaming pipeline picutre
- Adding future version pipeline picutre
- Changes to README for pipeline picture
- Changes to README for pipeline picture
- Changes to README for pipeline picture
- initial trial with kallisto exporten t2g
- Fixing small bug in check_samplesheet
- Fix prettier block.
- adding gene names to features.tsv for kallisto aligner
- fix skip_fastqc channel generations
- added fix info to changelog
- Add Florian Heyl
- Add tiny mermaid diagram for existing stuff
- change parameter to boolean
- Adding note for options
- adding enabled: params.save_reference to modules that should be affected by it
- update changelog
- added cellranger exports to 10x counts
- finished star and cellranger
- added export_mtx paramter
- black formatter
- fixed cellranger stub
- prettier formatting
- fixed other stub
- added changelog
- Update bin/cellranger_mtx_to_h5ad.py
- removed --export_mtx and improved kallisto gene name mapping
- exporting versions in mtx_to_h5ad and unified cellranger w other aligners
- run black formatter & prettier
- Update CHANGELOG.md
- Update bin/mtx_to_h5ad.py
- added task_process name to versions & improved code organization
- add UniverSC subworkflow as aligner option
- set up configuration for UniverSC aligner step
- add UniverSC to documentation
- rename duplicate parameter for index (lint passes)
- parameter input checks pass with technology as string not file
- update GitHub Actions to test UniverSC aligner
- Template update for nf-core/tools version 2.7.1
- Template update for nf-core/tools version 2.7.2
- Fix 196
- Updated black
- Adjust simpleaf to use latest simpleaf recipes / containers
- Also update alevinqc
- Update modules properly
- Modules update
- Update schema to make lint pass
- Another two [skip_ci]
- Revert to 1.12.1
- update to call UniverSC module https://github.com/nf-core/modules/pull/1706
- remove trailing spaces from the documentation
- correct syntax errors in UniverSC subworkflow
- correct syntax errors in UniverSC subworkflow
- bundle universc module into pipeline
- update docs to pass linting
- define universc technology variable
- update index for universc
- Use conda proper packages too
- Update modules, fix issues with pyroe hopefully
- Duplicate entries due to merge
- Typo :-(
- Build bump
- require txp2gene file when a salmon_index supplied
- modify assertion error to reflect changes
- tidying up the assertion
- fix ci issue
- fix ci syntax issue
- Re-add conda.enabled
- Fix Seurat container
- Update Changelog
- Update modules
- Fix linting
- try fix linting
- bump version
- Fix README/docs



## PR checklist

- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/scrnaseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/scrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
